ciongke/Makefile

```diff
@@ -0,0 +1,167 @@
+# - hook-deployment updates the deployment.
+# - hook-service create the hook service.
+# - test-image builds and pushes the test-image.
+
+create-cluster:
+	gcloud -q container --project "$(PROJECT)" clusters create ciongke --zone "$(ZONE)" --machine-type "$(MACHINE_TYPE)" --scope "https://www.googleapis.com/auth/compute","https://www.googleapis.com/auth/devstorage.full_control","https://www.googleapis.com/auth/logging.write","https://www.googleapis.com/auth/servicecontrol","https://www.googleapis.com/auth/service.management" --num-nodes "$(NUM_NODES)" --network "default" --enable-cloud-logging --enable-cloud-monitoring
+	gsutil mb -p "$(PROJECT)" "gs://$(RESULTS_BUCKET)"
+	gsutil acl ch -u AllUsers:R "gs://$(RESULTS_BUCKET)"
+	gsutil defacl ch -u AllUsers:R "gs://$(RESULTS_BUCKET)"
+	gsutil mb -p "$(PROJECT)" "gs://$(SOURCE_BUCKET)"
+	gsutil acl ch -u AllUsers:R "gs://$(SOURCE_BUCKET)"
+	gsutil defacl ch -u AllUsers:R "gs://$(SOURCE_BUCKET)"
+	kubectl create secret generic hmac-token --from-file=hmac=$(HOOK_SECRET_FILE)
+	kubectl create secret generic oauth-token --from-file=oauth=$(OAUTH_SECRET_FILE)
+	make test-pr-image --no-print-directory
+	make hook-image --no-print-directory
+	make hook-deployment --no-print-directory
+	make hook-service --no-print-directory
+	@echo -n "Waiting for loadbalancer ingress "; while [[ "$$(kubectl get svc hook -o=json | jq -r .status.loadBalancer.ingress[0].ip)" == "null" ]]; do echo -n "."; sleep 5; done; echo " Done"
+	@echo "Webhook address: http://$$(kubectl get svc hook -o=json | jq -r .status.loadBalancer.ingress[0].ip):8888/"
+
+delete-cluster:
+	@echo "Deleting cluster in $(PROJECT) in 5 seconds..."; sleep 5
+	gcloud -q container --project "$(PROJECT)" clusters delete ciongke
+	gsutil -mq rm -r "gs://$(RESULTS_BUCKET)"
+	gsutil -mq rm -r "gs://$(SOURCE_BUCKET)"
+
+clean:
+	rm cmd/test-pr/test-pr cmd/hook/hook
+
+.PHONY: create-cluster delete-cluster clean
+
+test-pr-image:
+	CGO_ENABLED=0 go build -o cmd/test-pr/test-pr github.com/kubernetes/test-infra/ciongke/cmd/test-pr
+	docker build -t "gcr.io/$(PROJECT)/test-pr:$(TEST_PR_VERSION)" cmd/test-pr
+	gcloud docker push "gcr.io/$(PROJECT)/test-pr:$(TEST_PR_VERSION)"
+
+hook-image:
+	CGO_ENABLED=0 go build -o cmd/hook/hook github.com/kubernetes/test-infra/ciongke/cmd/hook
+	docker build -t "gcr.io/$(PROJECT)/hook:$(HOOK_VERSION)" cmd/hook
+	gcloud docker push "gcr.io/$(PROJECT)/hook:$(HOOK_VERSION)"
+
+hook-deployment:
+	@echo "$${hook_deployment}" | kubectl apply -f -
+
+hook-service:
+	@echo "$${hook_service}" | kubectl create -f -
+
+.PHONY: test-pr-image hook-image hook-deployment hook-service
+
+test-image:
+	docker build -t "gcr.io/$(PROJECT)/test-image:$(TEST_IMAGE_VERSION)" test-image
+	gcloud docker push "gcr.io/$(PROJECT)/test-image:$(TEST_IMAGE_VERSION)"
+.PHONY: test-image
+
+# We need to replace project and version and so on in these YAML definitions,
+# and there's not really a great way to do that.
```

Comment:
Ehh, I'm not sure about that. For this deployment we should need 8 replacements, and the list may grow. It may be better to have the blob in a shell script that inherits variables from the makefile, though.